Tests/UpstreamIntegrationTests/UpstreamIntegrationTests.swift

```diff
@@ -69,52 +66,495 @@ class UpstreamIntegrationTests: TestBase {
+
+        let experienceEvent = ExperienceEvent(xdm: ["xdmtest": "data"],
+                                              data: ["data": ["test": "data"]])
+        
+        Edge.sendEvent(experienceEvent: experienceEvent)
+        
+        assertExpectedEvents(ignoreUnexpectedEvents: true)
+    }
+    
+    func testSendEventx2_doesNotReceivesErrorEvent() {
+        // Setup
+        sendEventAndResetExpectations()
+        
+        expectEdgeEventHandle(expectedHandleType: TestConstants.EventSource.LOCATION_HINT_RESULT, expectedCount: 1)
+        expectEdgeEventHandle(expectedHandleType: TestConstants.EventSource.STATE_STORE, expectedCount: 1)
+        
+        let experienceEvent = ExperienceEvent(xdm: ["xdmtest": "data"], data: ["data": ["test": "data"]])
 
-        let expected = getAnyCodable(expectedJSON)!
+        // Test
+        Edge.sendEvent(experienceEvent: experienceEvent)
+        
+        // Verify
+        assertExpectedEvents(ignoreUnexpectedEvents: true)
+        
+        let errorEvents = getEdgeEventHandles(expectedHandleType: TestConstants.EventSource.ERROR_RESPONSE_CONTENT)
+        XCTAssertEqual(0, errorEvents.count)
+    }
+    
+    func testSendEventx2_receivesExpectedLocationHintEventHandle() {
+        // Setup
+        expectEdgeEventHandle(expectedHandleType: TestConstants.EventSource.LOCATION_HINT_RESULT)
 
-        let resultEvents = getDispatchedEventsWith(type: TestConstants.EventType.EDGE, source: "locationHint:result")
-        XCTAssertEqual(1, resultEvents.count)
-        guard let locationHintEvent = resultEvents.first else {
-            XCTFail("No valid location hint event found")
+        let experienceEvent = ExperienceEvent(xdm: ["xdmtest": "data"],
+                                              data: ["data": ["test": "data"]])
+        
+        Edge.sendEvent(experienceEvent: experienceEvent)
+        
+        // Extract location hint from Edge Network location hint response event
+        guard let locationHintResult = getLastLocationHintResultValue() else {
+            XCTFail("Unable to extract valid location hint from location hint result event handle.")
             return
         }
+        
+        // Reset all test expectations
+        networkService.reset()
+        resetTestExpectations()
+        
+        // Set actual testing expectations
+        expectEdgeEventHandle(expectedHandleType: TestConstants.EventSource.LOCATION_HINT_RESULT, expectedCount: 1)
+        
+        // Test
+        // 2nd event
+        Edge.sendEvent(experienceEvent: experienceEvent)
+        
+        // Verify
+        // Verify location hint consistency between 1st and 2nd event handles
+        let expectedLocationHintJSON = #"""
+        {
+          "payload": [
+            {
+              "ttlSeconds" : 123,
+              "scope" : "EdgeNetwork",
+              "hint" : "\#(edgeLocationHint?.rawValue ?? locationHintResult)"
```

Comment:
Same here, this should always be locationHintResult. And you can add an extra check for initial value: 
if (edgeLocationHint != nil) {
    XCTAssertEqual(edgeLocationHint?.rawValue, locationHintResult)
}